apps/tx-builder/src/components/forms/AddNewTransactionForm.tsx

```diff
@@ -0,0 +1,137 @@
+import { Text, Title, GenericModal, ModalFooterConfirmation, Button } from '@gnosis.pm/safe-react-components';
+import { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { toChecksumAddress, toWei, fromWei } from 'web3-utils';
+
+import { ContractInterface } from '../../hooks/useServices/interfaceRepository';
+import { ProposedTransaction } from '../../typings/models';
+import { encodeToHexData, getTxDescription, isValidAddress } from '../../utils';
+import { ModalBody } from '../ModalBody';
+import SolidityForm, {
+  CONTRACT_METHOD_INDEX_FIELD_NAME,
+  CONTRACT_VALUES_FIELD_NAME,
+  HEX_ENCODED_DATA_FIELD_NAME,
+  SolidityFormValuesTypes,
+  TOKEN_INPUT_NAME,
+  TO_ADDRESS_FIELD_NAME,
+} from './SolidityForm';
+
+type AddNewTransactionFormProps = {
+  contract: ContractInterface | null;
+  to: string;
+  transactions: ProposedTransaction[];
+  onAddTransaction: (transaction: ProposedTransaction) => void;
+  onRemoveTransaction: (index: number) => void;
+  onSubmitTransactions: () => void;
+  networkPrefix: undefined | string;
+  getAddressFromDomain: (name: string) => Promise<string>;
+  nativeCurrencySymbol: undefined | string;
+};
+
+const AddNewTransactionForm = ({
+  transactions,
+  onAddTransaction,
+  onSubmitTransactions,
+  onRemoveTransaction,
+  contract,
+  to,
+  networkPrefix,
+  getAddressFromDomain,
+  nativeCurrencySymbol,
+}: AddNewTransactionFormProps) => {
+  const [showReviewModal, setShowReviewModal] = useState(false);
+  const showABIWarning = contract && !contract?.methods.length;
+
+  const initialFormValues = {
+    [TO_ADDRESS_FIELD_NAME]: isValidAddress(to) ? to : '',
+    [CONTRACT_METHOD_INDEX_FIELD_NAME]: '0',
+  };
+
+  function onSubmit(values: SolidityFormValuesTypes) {
```

Comment:
this function is defined with `function` keyword but the component is defined with an arrow function, let's stay consistent

Reply:
done!